packages/director/src/lib/hooks/utils.ts

```diff
@@ -1,24 +1,80 @@
 import {
+  EventFilter,
   HookEvent,
   HookWithCustomEvents,
   isGithubHook,
+  isSlackHook,
+  RunSummary,
 } from '@sorry-cypress/common';
 
 export function shouldHookHandleEvent(
   event: HookEvent,
-  hook: HookWithCustomEvents
+  hook: HookWithCustomEvents,
+  runSummary: RunSummary,
+  branch: string
 ) {
   if (isGithubHook(hook)) {
     return true;
   }
 
-  if (!hook.hookEvents || !hook.hookEvents.length) {
-    return true;
-  }
+  if (isSlackHook(hook)) {
+    if (
+      hook.slackEventFilter === EventFilter.ONLY_FAILED &&
+      runSummary.failures === 0 &&
+      runSummary.skipped === 0
+    ) {
+      return false;
+    }
 
-  if (hook.hookEvents.includes(event)) {
-    return true;
+    if (
+      (hook.slackEventFilter === EventFilter.ONLY_SUCCESSFUL &&
+        runSummary.failures > 0) ||
+      runSummary.skipped > 0
+    ) {
+      return false;
+    }
+
+    if (hook.slackBranchFilter && hook.slackBranchFilter.length > 0) {
+      const isBranchInFilter = !hook.slackBranchFilter
+        // Branch filter supports only '*' and '?' wildcard symbols, not full regex syntax,
+        // so first we escape all special symbols except '*' and '?'
+        // and then replace '*' with '.*' and '?' with '.' before passing this string to regex.
+        // We shouldn't warn about handling branch names having '*' or '?' symbols
+        // as such names are prohibited (see 'man git-check-ref-format')
+        .map((filter: string) =>
+          filter
+            .replace(/[\-\[\]\/\{\}\(\)\+\.\\\^\$\|]/g, '\\$&')
+            .replace(/\*/g, '.*')
+            .replace(/\?/g, '.')
+        )
+        // Then we just check all filters for a mach with a users' branch,
+        // and if all filters don't matched (returned -1)
+        // we return inverted value to set isBranchInFilter to 'false'
+        .every(
+          (filter: string) =>
+            branch.search(new RegExp(`^${filter}$`, 'i')) === -1
+        );
+
+      if (!isBranchInFilter) return false;
+    }
+
+    if (!hook.hookEvents || !hook.hookEvents.length) {
+      return true;
+    }
+
+    if (hook.hookEvents.includes(event)) {
+      return true;
+    }
   }
 
   return false;
```

Comment:
That becomes a bit bulky to read, would you mind please extracting to a separate function, smth like `shouldReportSlackHook`

Reply:
Totally agree, thank you. I have split it into 4 functions. Also I have covered all of them by unit tests 😎